### PR TITLE
Python: Support creating a DateLiteral from a date (#6120)

### PR DESCRIPTION
**Pull request by ddrinka**
* Renamed date_to_days to date_str_to_days

* Created new date_to_days to convert a Python date object to int days

* Contributes to workaround for #6120

### CODE REVIEW

**Review by Fokko — APPROVED** [review]
Thanks for adding support for the date, this makes sense to me to have in there.

File: python/pyiceberg/utils/datetime.py [hunk 1]
```diff
@@ -47,11 +47,16 @@ def micros_to_time(micros: int) -> time:
     return time(hour=hours, minute=minutes, second=seconds, microsecond=microseconds)
 
 
-def date_to_days(date_str: str) -> int:
+def date_str_to_days(date_str: str) -> int:
```
Comment (rdblue) [inline python/pyiceberg/utils/datetime.py:50]: Does this need to be renamed? Why not support both with the same name to avoid a breaking change?

Comment (ddrinka) [reply python/pyiceberg/utils/datetime.py:50]: My thought process was that this API is still pre 1.0 and the existing name was misleading, so better to improve it now.  I have no preference either way though, so happy to adjust.  I'm not sure how to support both with the same name though?

**Review by rdblue — APPROVED** [review]

File: python/pyiceberg/utils/datetime.py (hunk 1 above)
Comment (rdblue) [reply python/pyiceberg/utils/datetime.py:50]: Yeah, it probably isn't used anywhere. You're right: better to clean it up.

**Comment by rdblue** [comment]
Thanks, @ddrinka! I merged this.

**Pull request merged**